aws-s3-transfer-manager/src/operation/upload/service.rs

```diff
@@ -69,7 +70,19 @@ pub(super) fn upload_part_service(
        + Send {
     let svc = service_fn(upload_part_handler);
     let concurrency_limit = ConcurrencyLimitLayer::new(ctx.handle.scheduler.clone());
-    ServiceBuilder::new().layer(concurrency_limit).service(svc)
+
+    let svc = ServiceBuilder::new()
+        .layer(concurrency_limit)
```

Comment:
A "hedged" request should count towards to concurrency limit. If this is hard to do, let's do it in a followup task.

Even better would be if hedged requests were lower priority, and only got a ticket when there's no other work to do. Since they really only help us if we're done with a workload and all that's left is the slow requests. Otherwise they're just stealing from another request that would be more useful.

Reply:
I have looked into this, and it seems like the hedged requests can bypass the concurrency_limit layer. Reorganizing the layers had no effect. I have added a TODO to fix this for now since I want to get this merged and see if there is any difference in benchmarks.

Another interesting thing I noticed is that we are hedging very few requests, like for a 30GB upload with 125 concurrency, we only hedged less 5-15 requests. We can revisit the numbers once we have more tracing/metrics in place.